src/outputs/parthenon_hdf5.cpp

```diff
@@ -225,232 +227,213 @@ void PHDF5Output::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) {
+    local_count[1] = nx3;
+    local_count[2] = nx2;
+    local_count[3] = nx1;
+    local_count[4] = 1;
+
+    global_count[1] = nx3;
+    global_count[2] = nx2;
+    global_count[3] = nx1;
+    global_count[4] = 1;
+
+    H5S const local_DSpace = H5S::FromHIDCheck(H5Screate_simple(5, local_count, NULL));
+    H5S const global_DSpace = H5S::FromHIDCheck(H5Screate_simple(5, global_count, NULL));
+
+    // while we could do this as n variables and load all variables for
+    // a block at one time, this is memory-expensive.  I think it is
+    // well worth the multiple iterations through the blocks to load up
+    // one variable at a time.  Besides most of the time will be spent
+    // writing the HDF5 file to disk anyway...
+    // If I'm wrong about this, we can always rewrite this later.
+    // Sriram
+
+    // this is a stupidly complicated multi-pass through the variable
+    // list, but again will revisit when the time comes to redo
+    for (auto &vwrite : ciX.vars) { // for each variable we write
+      const std::string vWriteName = vwrite->label();
+      hid_t vLocalSpace, vGlobalSpace;
+      H5S vLocalSpaceNew, vGlobalSpaceNew;
+      const hsize_t vlen = vwrite->GetDim(4);
+      local_count[4] = global_count[4] = vlen;
+
+      if (vlen == 1) {
+        vLocalSpace = local_DSpace;
+        vGlobalSpace = global_DSpace;
+      } else {
+        vLocalSpace = vLocalSpaceNew =
+            H5S::FromHIDCheck(H5Screate_simple(5, local_count, NULL));
+        vGlobalSpace = vGlobalSpaceNew =
+            H5S::FromHIDCheck(H5Screate_simple(5, global_count, NULL));
+      }
+
+      hsize_t index = 0;
+      for (auto &pmb : pm->block_list) { // for every block1
+        auto ci =
+            ContainerIterator<Real>(pmb->real_containers.Get(), output_params.variables);
+        for (auto &v : ci.vars) {
+          std::string name = v->label();
+          if (name.compare(vWriteName) == 0) {
+            // hsize_t index = pmb->lid * varSize * vlen;
+            auto v_h = v->data.GetHostMirrorAndCopy();
+            LOADVARIABLEONE(index, tmpData, v_h, out_ib.s, out_ib.e, out_jb.s, out_jb.e,
+                            out_kb.s, out_kb.e, vlen);
+            break;
+          }
         }
       }
-    }
-    // write dataset to file
-    WRITEH5SLAB2(vWriteName.c_str(), tmpData, file, local_start, local_count, vLocalSpace,
-                 vGlobalSpace, property_list);
-    if (vlen > 1) {
-      H5Sclose(vLocalSpace);
-      H5Sclose(vGlobalSpace);
+      // write dataset to file
+      WRITEH5SLAB2(vWriteName.c_str(), tmpData.data(), file, local_start, local_count,
+                   vLocalSpace, vGlobalSpace, property_list);
     }
   }
-  // close data spaces
-  H5Sclose(local_DSpace);
-  H5Sclose(global_DSpace);
-
-#ifdef MPI_PARALLEL
-  /* release the file access template */
-  ierr = H5Pclose(acc_file);
-  ierr = MPI_Info_free(&FILE_INFO_TEMPLATE);
-#endif
-
-  H5Pclose(property_list);
-  H5Fclose(file);
```

Comment:
Same as below. No need to close here any more?

Reply:
Yeah, see other thread.